agent/engine/docker_task_engine.go

```diff
@@ -302,9 +303,9 @@ func (engine *DockerTaskEngine) monitorExecAgentRunning(ctx context.Context,
 	_, err = engine.execCmdMgr.RestartAgentIfStopped(ctx, engine.client, task, c, dockerID)
 	if err != nil {
 		seelog.Errorf("Task engine [%s]: Failed to restart ExecCommandAgent Process for container [%s]: %v", task.Arn, dockerID, err)
-		// TODO: [ecs-exec] call control plane to report ExecCommandAgent FAIL here
 	}
-	// TODO: [ecs-exec] call control plane to report ExecCommandAgent RESTARTED/UNKNOWN? if RestartIfStopped returns true
+	seelog.Infof("restarted")
+	mTask.emitContainerEvent(mTask.Task, c, "")
```

Comment:
This should be done only if the result of `RestartAgentIfStopped` is  `execcmd.Restarted`.
See https://github.com/aws/amazon-ecs-agent/blob/feature/ecs_exec/agent/engine/execcmd/manager.go#L47

Reply:
We need to emit an event if we succeed or fail to restart.  (succeed means STOPPED->RUNNING or RUNNING->RUNNING while fail means RUNNING->STOPPED or STOPPED-> STOPPED.)  
I'm adding a comment to clarify:
+       // whether we restarted or failed to restart, we'll want to emit a state change event
+       // redundant state change events like RUNNING->RUNNING are allowed